app/controllers/submit.js

```diff
@@ -358,7 +358,7 @@ export default Ember.Controller.extend(BasicsValidations, NodeActionsMixin, Tagg
                 });
             });
 
-            Ember.RSVP.all(promisesArray).then(() => {
+            Ember.RSVP.allSettled(promisesArray).then(() => {
```

Comment:
Small heads up- this may affect any error handling you do. Since failure results are summarized in objects, IIRC- if a request fails it still kicks you into the `.then` rather than `catch`.

Relevant if you want to add a future mechanism to warn the user that deletion/ cleanup failed; all vs allsettled don't appear to be drop-in replacements. http://emberjs.com/api/classes/RSVP.html#method_allSettled

Though I'm pretty fried, so rather than pretending to be 20, it's off to bed I go. Thanks for the hard work!